rskj-core/src/main/java/co/rsk/rest/RestServerDispatcher.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.rest;
+
+import co.rsk.rest.modules.RestModule;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class RestServerDispatcher {
+    private static final Logger logger = LoggerFactory.getLogger(RestServerDispatcher.class);
+
+    private static final String NOT_FOUND = "Not Found";
+
+    private final List<RestModule> moduleList;
+
+    public RestServerDispatcher(List<RestModule> moduleList) {
+        Objects.requireNonNull(moduleList, "Module List can not be null");
+        this.moduleList = Collections.unmodifiableList(moduleList);
+    }
+
+    public DefaultFullHttpResponse dispatch(HttpRequest request) throws URISyntaxException {
+
+        String uri = new URI(request.getUri()).getPath();
+
+        RestModule restModule = moduleList.stream()
```

Comment:
On thing, we are only filtering by path but not by method (GET,PUT...) and this is done internally in the module. If we decide to return 404 if the method doesn't match then we can filter directly here. and avoid any other logic.

Reply:
The path being checked here is only the start bit of it, since that part defines what's the module that has to take care of the incoming request. Method and full path are taken care by the module itself.